### PR TITLE
Ensuring deleted cookies are recognized by the browser by ensuring the MaxAge within CookieOptions is set to null.

### LINKED ISSUE #52159: ResponseCookies.Delete(key, cookieOptions) doesn't delete cookie if the caller provides a non-null max-age
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

I've run into a subtle change in behavior in `ResponseCookies.Delete(key, cookieOptions)` where the cookie isn't deleted if the caller provides a non-null MaxAge.

## .NET 6 behavior

[Relevant line of code](https://github.com/dotnet/aspnetcore/blob/release/6.0/src/Http/Http/src/Internal/ResponseCookies.cs#L204C15-L204C15)

```cs
Append(key, string.Empty, new CookieOptions
{
    Path = options.Path,
    Domain = options.Domain,
    Expires = DateTimeOffset.UnixEpoch,
    Secure = options.Secure,
    HttpOnly = options.HttpOnly,
    SameSite = options.SameSite
});
```

A new `CookieOptions` object is created, copying the values for Path, Domain, Secure, HttpOnly, and SameSite. Expires is forced to `DateTimeOffset.UnixEpoch`. MaxAge is left with its default value of `null`.

**End result:** cookie is deleted regardless of what the caller provides for Expires and MaxAge.

## .NET 8 behavior

[Relevant line of code](https://github.com/dotnet/aspnetcore/blob/526417ee2be43995df9103ea3b2738ef766cd766/src/Http/Http/src/Internal/ResponseCookies.cs#L163C11-L163C11)
```cs
Append(key, string.Empty, new CookieOptions(options)
{
    Expires = DateTimeOffset.UnixEpoch,
});
```

The new `CookieOptions` is created using the copy constructor. This copies all the properties, including MaxAge.

**End result:** if the caller happens to provide a non-null value of MaxAge the cookie will not be deleted. Or at least, this is the behavior I'm seeing in Chrome 119.0.6045.160. I believe browsers give `max-age` precedence over `expires`. 

I'm running into this because I have a cached `CookieOptions` that I'm using for both the set and the delete (mostly to ensure the same Path is used for both calls). But as a side effect, I'm sending a non-null MaxAge to the delete and the cookie is not being deleted.

## Workaround
Callers can ensure that MaxAge is `null` before calling `Delete`.

### Expected Behavior

The cookie should be deleted even if the caller happens to provide a non-null MaxAge.

### Steps To Reproduce

_No response_

### Exceptions (if any)

_No response_

### .NET Version

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by MattyLeslie**
**Reset MaxAge to Null on Cookie Deletion**

**Summary:**
This PR aims to ensure reliable cookie deletion by setting MaxAge to null in the CookieOptions used by ResponseCookies.Delete, addressing inconsistencies identified in issue #52159.

**Description:**
To resolve inconsistencies in cookie deletion across browsers, this update modifies the ResponseCookies.Delete method to reset MaxAge to null. Previously, retained MaxAge values could prevent browsers from recognizing that cookies were meant to be deleted.

**Technical Changes:**

Enforced MaxAge = null in CookieOptions during cookie deletion..

Fixes
#52159 by standardizing cookie deletion behavior across all browsers.

### CODE REVIEW

**Comment by amcasey** [comment]
CI failure looks like network flakiness.  Re-running.

File: src/Http/Http/src/Internal/ResponseCookies.cs [hunk 1]
```diff
@@ -163,6 +163,7 @@ public void Delete(string key, CookieOptions options)
         Append(key, string.Empty, new CookieOptions(options)
         {
             Expires = DateTimeOffset.UnixEpoch,
+            MaxAge = null,
```
Comment (amcasey) [inline src/Http/Http/src/Internal/ResponseCookies.cs:166]:
```suggestion
            MaxAge = null, // Some browsers require this (https://github.com/dotnet/aspnetcore/issues/52159)
```

**Comment by amcasey** [comment]
I like it!  Do we need to update [here](https://github.com/dotnet/aspnetcore/blob/561f135501c5cbf1bb6456d494669a35a088b493/src/Shared/ChunkingCookieManager/ChunkingCookieManager.cs#L306) and [here](https://github.com/dotnet/aspnetcore/blob/561f135501c5cbf1bb6456d494669a35a088b493/src/Shared/ChunkingCookieManager/ChunkingCookieManager.cs#L333) as well?

**Comment by MattyLeslie** [comment]
Thanks for the review @amcasey, I believe the `ChunkingCookieManager.cs` will need to make use of the same logic when deleting chunks, I have updated the relevant areas and added the recommended comments.

**Review by amcasey — APPROVED** [review]
Thanks!

**Pull request merged**